cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -813,27 +844,34 @@ public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration
-     */
-    private void addMetricReporters(KafkaConfiguration userConfig, boolean injectCcMetricsReporter, boolean injectJmxReporter) {
-        if (injectJmxReporter)  {
-            if (userConfig.getConfigOption("metric.reporters") != null && !userConfig.getConfigOption("metric.reporters").contains(KAFKA_JMX_REPORTER_CLASS)) {
-                userConfig.setConfigOption("metric.reporters", userConfig.getConfigOption("metric.reporters") + "," + KAFKA_JMX_REPORTER_CLASS);
-            } else {
-                userConfig.setConfigOption("metric.reporters", KAFKA_JMX_REPORTER_CLASS);
-            }
+     * @param userConfig The user configuration to which the metric reporters will be added.
+     * @param injectCcMetricsReporter Flag indicating whether to inject the Cruise Control Metrics Reporter.
+     * @param injectKafkaJmxReporter  Inject the JMX Reporter into the configuration
+     * @param injectStrimziMetricsReporter Flag indicating whether to inject the Strimzi Metrics Reporter.
+     */
+    private void maybeAddMetricReporters(KafkaConfiguration userConfig, boolean injectCcMetricsReporter, boolean injectKafkaJmxReporter, boolean injectStrimziMetricsReporter) {
+        if (injectCcMetricsReporter) {
+            createOrAddListConfig(userConfig, "metric.reporters", CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
+        }
+        if (injectKafkaJmxReporter) {
+            createOrAddListConfig(userConfig, "metric.reporters", "org.apache.kafka.common.metrics.JmxReporter");
         }
+        if (injectStrimziMetricsReporter) {
+            createOrAddListConfig(userConfig, "metric.reporters", "io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter");
+        }
+    }
 
-        if (injectCcMetricsReporter)  {
-            if (userConfig.getConfigOption("metric.reporters") != null && !userConfig.getConfigOption("metric.reporters").contains(CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER)) {
-                userConfig.setConfigOption("metric.reporters", userConfig.getConfigOption("metric.reporters") + "," + CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-            } else {
-                userConfig.setConfigOption("metric.reporters", CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-            }
+    /**
+     * Adds the Yammer kafka.metrics.reporters to the user configuration if the Strimzi Metrics Reporter is enabled.
+     *
+     * @param userConfig The user configuration to which the Yammer metrics reporter will be added.
+     * @param injectStrimziMetricsReporter Flag indicating whether to inject the Strimzi Metrics Reporter.
+     */
+    private void maybeAddYammerMetricsReporters(KafkaConfiguration userConfig, boolean injectStrimziMetricsReporter) {
+        if (injectStrimziMetricsReporter) {
+            createOrAddListConfig(userConfig, "kafka.metrics.reporters", "io.strimzi.kafka.metrics.YammerPrometheusMetricsReporter");
```

Comment:
Maybe we can have a constant for "io.strimzi.kafka.metrics.YammerPrometheusMetricsReporter" within the `StrimziMetricsReporterModel` class because it's anyway set in related to the Strimzi Metrics Reporter.